onnxruntime/core/providers/cuda/nn/pool.cc

```diff
@@ -165,7 +167,7 @@ Status Pool<T, PoolType, NHWC>::ComputeInternal(OpKernelContext* context) const
     pads.assign(kernel_shape.size(), 0);
     strides.assign(kernel_shape.size(), 1);
   }
-  auto out_channel = NHWC ? x_shape[3] : x_shape[1];
+  auto out_channel = NHWC ? x_shape[x_dims.size() - 1] : x_shape[1];
```

Comment:
FYI- Not sure if this needs to be addressed here (lines 185...188 below) to make the NHWC 1-D tests pass (https://github.com/microsoft/onnxruntime/pull/19780#discussion_r1513382630)

Reply:
the 1d indexed test passes. I'll add the new MaxPool1D_case3 as well and do more testing.

I have a pending change for GlobalMaxPool as well which uncovers one more issue in the global pool logic as well as in the NHWC logic in `Status CudnnTensor::Set(gsl::span<const int64_t> input_dims, cudnnDataType_t dataType, bool is_nhwc)`

For the NHWC case the following logic is used:
```
  if (is_nhwc) {
    std::swap(dims[1], dims[rank - 1]);
    std::swap(strides[1], strides[rank - 1]);
  }
```

Unless I confused myself today searching for a bug this means NHWC permutes to NCWH which is incorrect - yet it produces the correct result for all code which currently is using the logic except for 3d global maxpooling.

I changed the logic to the code below which works perfectly fine for NHWDC, but fails for NHWC. Any hint about how to solve the knot in my head is appreciated. I'll dig deeper tomorrow what's going on here.

```
  if (is_nhwc) {
    dims[0] = input_dims[0]; // take N
    dims[1] =input_dims[rank-1]; // take C
    for (i = 1;i < rank-1;++i) { // take HW(D)
        dims[i+1] = input_dims[i]);
    }
  }
```